### PR TITLE
[flare] Log permissions on collected files

### PR DESCRIPTION
**Pull request by olivielpeau**
The permissions are collected in a files_permissions.log file.
All the files are added to the tar through an `_add_tar_file` method.

The generated `files_permissions.log` file looks like the following:

```
File path                                          | mode  | owner      | group
/var/log/datadog/collector.log                     | 0644  | dd-agent   | nogroup
/var/log/datadog/forwarder.log                     | 0644  | dd-agent   | nogroup
/var/log/datadog/dogstatsd.log                     | 0644  | dd-agent   | nogroup
/var/log/datadog/jmxfetch.log                      | 0644  | dd-agent   | nogroup
/var/log/datadog/supervisord.log                   | 0644  | dd-agent   | root
/etc/dd-agent/datadog.conf                         | 0644  | dd-agent   | root
/etc/dd-agent/supervisor.conf                      | 0644  | dd-agent   | root
/etc/dd-agent/conf.d/tomcat.yaml                   | 0644  | dd-agent   | root
[...]
```

### CODE REVIEW

File: utils/flare.py [hunk 1]
```diff
@@ -3,9 +3,12 @@
 import cStringIO as StringIO
 from functools import partial
 import glob
+import grp
 import logging
-import os.path
+import os
+import pwd
```
Comment (degemer) [inline utils/flare.py]: Is it possible to `import` `grp` and `pwd` on Windows without error ? (I saw that they were Unix-only modules)

File: utils/flare.py [hunk 2]
```diff
@@ -158,6 +167,15 @@ def _init_tarfile(self):
...
         self._tar = tarfile.open(self._tar_path, 'w:bz2')
 
+    # Create a file to log permissions on collected files and write header line
+    def _init_permissions_file(self):
+        self._permissions_file = tempfile.NamedTemporaryFile(mode='w', prefix='dd', delete=False)
+        if Platform.is_unix():
+            self._permissions_file_format = "{0:50} | {1:5} | {2:10} | {3:10}\n"
+            self._permissions_file.write(self._permissions_file_format.format("File path", "mode", "owner", "group"))
+        else:
+            "Not implemented: file permissions are only logged on Unix platforms"
```
Comment (degemer) [inline utils/flare.py]: A `log.` or `print` missing ?

File: utils/flare.py (hunk 1 above)
Comment (olivielpeau) [inline utils/flare.py]: Hmm I haven't checked but you're probably right, I'll change this.

File: utils/flare.py (hunk 2 above)
Comment (olivielpeau) [inline utils/flare.py]: Oops, good catch, I wanted to write that in the file.

**Comment by degemer** [comment]
Once you have fixed the last _esthetic_ comments (file name and separation between title and content), merge it (but cancel the build before). :tada:

**Comment by olivielpeau** [comment]
Thanks @degemer! I've changed the file name to `permissions.log` and added the separation you mentioned :gem: 

Merging.

**Pull request merged**